java/kotlin-extractor/src/main/kotlin/KotlinUsesExtractor.kt

```diff
@@ -212,10 +253,12 @@ open class KotlinUsesExtractor(
         val extractClass = substituteClass ?: c
 
         // `KFunction1<T1,T2>` is substituted by `KFunction<T>`. The last type argument is the return type.
+        // Similarly Function23 and above get replaced by kotlin.jvm.functions.FunctionN with only one type arg, the result type.
         // References to SomeGeneric<T1, T2, ...> where SomeGeneric is declared SomeGeneric<T1, T2, ...> are extracted
         // as if they were references to the unbound type SomeGeneric.
         val extractedTypeArgs = when {
-            c.symbol.isKFunction() && typeArgs != null && typeArgs.isNotEmpty() -> listOf(typeArgs.last())
+            extractClass.symbol.isKFunction() && typeArgs != null && typeArgs.isNotEmpty() -> listOf(typeArgs.last())
+            extractClass.fqNameWhenAvailable == FqName("kotlin.jvm.functions.FunctionN") && typeArgs != null && typeArgs.isNotEmpty() -> listOf(typeArgs.last())
```

Comment:
Is there a test case for this? How come we didn't see this issue before? We have tests that cover big-arity functions.

Reply:
I can add one -- I think we simply didn't notice or care that the type extracted was `FunctionN<FirstArgumentType>` not `FunctionN<ReturnType>`. It's probably not a critical bug but rather a weird surprise waiting for someone down the line.